src/core/fetchRequest.js

```diff
@@ -54,9 +114,11 @@ const requestFactory = (url, options) => {
             });
     }
 
+    const httpRequestImplementation = options.httpRequestImplementation === 'xhr' ? xhr : fetch;
```

Comment:
Normally we used this [core/fetchRequest](https://github.com/oat-sa/tao-core-sdk-fe/blob/master/src/core/fetchRequest.js) for fetch, and the older [core/request](https://github.com/oat-sa/tao-core-sdk-fe/blob/master/src/core/request.js) for xmlHttpRequest (our module probably doesn't support the progress options you want to add, but it could make more sense to add them there).

Reply:
@oatymart these two modules have too different API, can I introduce a breaking change and update old request module API so it matches fetchRequest module API (like same options, same return types)?
This will allow for less code duplication

Reply:
I probably wouldn't recommend a breaking change on `core/request`, because it is used in many places (in CG TAO at least). Don't you see it as possible to add the optional `onProgress` callbacks in a non-breaking way?

Reply:
@oatymart 
<img width="878" alt="image" src="https://github.com/oat-sa/tao-core-sdk-fe/assets/19402107/031ca303-3113-4705-ae1c-d0a0f61df013">
There's also another problem with using the old request module - it depends on jquery, and in order to use this in TAO Portal for one single request - we'll have to install it. There's a possibility to use XHR w/o doing it in the tao-core, but whole logic with updating the expired token and error handling has to be completely repeated there, which is why I thought implementing xhr as optional request implementation in `fetchRequest` was ok